solver/src/driver.rs

```diff
@@ -179,12 +185,35 @@ impl Driver {
         settlements.sort_by_cached_key(|(_, settlement)| {
             Reverse(settlement.objective_value(&estimated_prices))
         });
+        let settle_orders_older_than =
+            chrono::offset::Utc::now() - chrono::Duration::from_std(self.min_order_age).unwrap();
         for (solver, settlement) in settlements {
-            info!("{} computed {:?}", solver, settlement);
+            tracing::info!("{} computed {:?}", solver, settlement);
+
             if settlement.trades.is_empty() {
-                info!("Skipping empty settlement");
+                tracing::info!("Skipping empty settlement");
+                continue;
+            }
+
+            // If all orders are younger than self.min_order_age skip settlement. Orders will still
+            // be settled once they have been in the order book for longer. This makes coincidence
+            // of wants more likely.
+            let should_be_settled_immediately = settlement.trades.iter().any(|trade| {
+                let meta = &orders
+                    .iter()
+                    .find(|order| order.order_creation == trade.order)
```

Comment:
This is optimal from a big-o perspective but we know that the number of possible trades in a single settlement won't be huge and it makes the code easier than for example creating a hash map of order uid.